src/components/application_manager/test/commands/mobile/set_audio_streaming_indicator_test.cc

```diff
@@ -0,0 +1,514 @@
+                            const bool is_success) {
+    EXPECT_EQ((*msg)[am::strings::msg_params][am::strings::success].asBool(),
+              is_success);
+    EXPECT_EQ((*msg)[am::strings::msg_params][am::strings::result_code].asInt(),
+              mobile_code);
+  }
+
+  void AppSetup(const bool is_navi,
+                const bool is_voice_communication,
+                const bool is_media) {
+    ON_CALL(*mock_app_, is_navi()).WillByDefault(Return(is_navi));
+    ON_CALL(*mock_app_, is_media_application()).WillByDefault(Return(is_media));
+    ON_CALL(*mock_app_, is_voice_communication_supported())
+        .WillByDefault(Return(is_voice_communication));
+  }
+
+  void MediaAppSetup() {
+    AppSetup(kIsNotNavi, kIsNotVoiceCommunication, kIsMedia);
+  }
+
+  void VoiceCommunicationAppSetup() {
+    AppSetup(kIsNotNavi, kIsVoiceCommunication, kIsNotMedia);
+  }
+
+  void NaviAppSetup() {
+    AppSetup(kIsNavi, kIsNotVoiceCommunication, kIsNotMedia);
+  }
+
+  void IncorrectAppSetup() {
+    AppSetup(kIsNotNavi, kIsNotVoiceCommunication, kIsNotMedia);
+  }
+
+  mobile_apis::AudioStreamingIndicator::eType GetIndicator(
+      MessageSharedPtr msg) {
+    return (static_cast<mobile_apis::AudioStreamingIndicator::eType>(
+        (*msg)[am::strings::msg_params][am::strings::audio_streaming_indicator]
+            .asInt()));
+  }
+
+  void TestCommandOnEventCall(
+      const mobile_apis::Result::eType mobile_code,
+      const bool is_success,
+      MessageSharedPtr msg_hmi_response,
+      utils::SharedPtr<SetAudioStreamingIndicatorRequest> cmd) {
+    const hmi_apis::Common_Result::eType hmi_code =
+        static_cast<hmi_apis::Common_Result::eType>(
+            (*msg_hmi_response)[am::strings::params][am::hmi_response::code]
+                .asInt());
+
+    Event event_ui(hmi_apis::FunctionID::UI_SetAudioStreamingIndicator);
+    event_ui.set_smart_object(*msg_hmi_response);
+
+    EXPECT_CALL(mock_message_helper_, HMIToMobileResult(hmi_code))
+        .WillOnce(Return(mobile_code));
+
+    MessageSharedPtr msg_mobile_response =
+        CatchMobileCommandResult(CallOnEvent(*cmd, event_ui));
+    VerifyCommandResults(msg_mobile_response, mobile_code, is_success);
+  }
+
+  NiceMock<MockHmiInterfaces> hmi_interfaces_;
+  MockMessageHelper& mock_message_helper_;
+  MockAppPtr mock_app_;
+};
+
+TEST_F(SetAudioStreamingIndicatorRequestTest,
+       Run_NotRegisteredApplication_APPLICATION_NOT_REGISTERED) {
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+
+  const mobile_apis::Result::eType mobile_response_code =
+      mobile_apis::Result::APPLICATION_NOT_REGISTERED;
+  const bool is_success = kIsNotSuccess;
+
+  ApplicationSharedPtr mock_app_empty;
+  ON_CALL(app_mngr_, application(kConnectionKey))
+      .WillByDefault(Return(mock_app_empty));
```

Comment:
@tkireva, this one looks important to this flow. Why do we don't expect this call, but check an argument?